time_test.go

```diff
@@ -0,0 +1,65 @@
+package pflag
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+)
+
+func setUpTime(t *time.Time, formats []string) *FlagSet {
+	f := NewFlagSet("test", ContinueOnError)
+	f.TimeVar(t, "time", time.Time{}, formats, "Time")
+	return f
+}
+
+func TestTime(t *testing.T) {
+	testCases := []struct {
+		input    string
+		success  bool
+		expected string
```

Comment:
Nit: I think I'd prefer if this was a `time.Time` rather than a string, to make it easier to add other test cases without having to dig into how comparisons are done and what format to specify dates in. But I'm not going to block on that :)

Reply:
Updated in c5ce22e836c6268eb270e8f28ae5e3729a27c82d